Dockerfile.dapper

```diff
@@ -1,6 +1,9 @@
-FROM ubuntu:20.04
+FROM ubuntu:22.04
 # FROM arm=armhf/ubuntu:20.04 arm64=arm64v8/ubuntu:20.04
 
+ENV DOCKER_VERSION=23.0.3
+ENV GO_VERSION=1.20.1
```

Comment:
Latest is 1.20.3

Reply:
Changed to `1.20.3`. The upcoming `1.20.4` will bring some security fixes, so we may want to bump that again - maybe as a follow-up PR, as it will be out later on this week or next.

Reply:
No longer relevant now the base image is `bci-base`, as the go version will be kept up to date automatically.